tests/ui/iter_next_slice.rs

```diff
@@ -0,0 +1,39 @@
+#![warn(clippy::iter_next_slice)]
+
+fn main() {
+    // test code goes here
+    let s = [1, 2, 3];
+    let v = vec![1, 2, 3];
+    let o = Some(5);
+    
+    s.iter().next();
+    // Should be replaced by s.get(0)
+    
+    // s.into_iter().next();
+    // Probaby can rely on the default `array_into_iter`/`clippy::into_iter_on_ref` lint
```

Comment:
This should not lint here, right? Please don't comment out this test.

Reply:
So for this lint's test case, I copied lights0123's suggestion here: https://github.com/rust-lang/rust-clippy/issues/5572#issuecomment-625927865

However since I wasn't sure that `into_iter().next()` is part of the issue's scope, I commented them and focus on `iter().next()` first. I could uncomment it, but should I put `array_into_iter`/`clippy::into_iter_on_ref` lint result into the `tests/ui/iter_next_slice.stderr`?

Reply:
I agree that `into_iter` should not be linted here, since this is already covered by another lint. Also `into_iter().next()` will return an owned value, while `get()` returns a reference.

You can either remove all `into_iter` tests or leave one in to check that this does not get linted by this lint (for one case the other lint output is fine). I fine with both.

Reply:
Okay, I think having one commented `into_iter` line is good, so I added it in the new commit and remove the rest.